packages/yarn-plugin-allow-scripts/test/index.spec.mts

```diff
@@ -0,0 +1,238 @@
+        env: {
+          ...process.env,
+          ...defaultOpts.env,
+          ...env,
+        }
+      },
+    )
+    return {
+      exitCode: result.status,
+      out: result?.stdout?.toString()?.trim(),
+      err: result?.stderr?.toString()?.trim(),
+      output: result?.output,
+      signal: result.signal,
+      error: result.error,
+    }
+  } catch(error) {
+    console.error(error);
+    t.fail(
+      `Failed calling '${args.join(' ')}': ${JSON.stringify(
+        {
+          cwd,
+          error,
+        },
+        undefined,
+        2
+      )}`
+    )
+    return {
+      error,
+      output: [],
+      exitCode: null,
+      signal: null,
+      out: '',
+      err: '',
+    };
+  }
+}
+
+test('install - errors when adding dependency with unspecified preinstall script', async (t: any) => {
+  const projectName = 'uninitialized';
+  const projectRoot = join(dirname(import.meta.url.replace(/^file:/, '')), 'projects', projectName);
+
+  try {
+    // delete any leftover test artifacts
+    rmSync(join(projectRoot, PACKAGE_JSON), { force: true })
+
+    // init project
+    const initRes = run(t, ['init', '-y'], projectRoot)
+    t.is(initRes.exitCode, 0);
+
+    // install the plugin
+    const importRes = run(t, ['plugin', 'import', normalize('../../../bundles/@yarnpkg/plugin-allow-scripts.js')], projectRoot)
+    t.is(importRes.err, '');
+    t.regex(importRes.out, /YN0000: Saving the new plugin in .yarn\/plugins\/@yarnpkg\/plugin-allow-scripts.cjs/);
+    t.is(importRes.exitCode, 0);
+
+    const addRes = run(t, ['add', '@lavamoat/preinstall-always-fail'], projectRoot)
+    t.regex(addRes.err, /allow-scripts detected attempted execution of unconfigured package script. \["@lavamoat\/preinstall-always-fail","preinstall"/);
+    t.is(addRes.exitCode, 1);
+
+    // trigger the plugin
+    const installRes = run(t, ['install'], projectRoot)
+    t.regex(installRes.out, /packages missing configuration:\n- @lavamoat\/preinstall-always-fail/);
+    t.regex(installRes.err, /@lavamoat\/allow-scripts has detected dependencies without configuration. explicit configuration required./);
+    t.is(installRes.exitCode, 1);
+  } finally {
+    cleanup(projectName);
+  }
+})
+
+test('install - allows adding and installing package with specified preinstall script', async (t: any) => {
+  const projectName = 'allowed';
+  const projectRoot = join(dirname(import.meta.url.replace(/^file:/, '')), 'projects', projectName)
+
+  try {
+    run(t, ['plugin', 'import', normalize('../../../bundles/@yarnpkg/plugin-allow-scripts.js')], projectRoot)
+    const addRes = run(t, ['add', '@lavamoat/preinstall-always-fail'], projectRoot)
+    t.is(addRes.exitCode, 0);
+
+    // trigger the plugin
+    const installRes = run(t, ['install'], projectRoot)
+    t.is(installRes.exitCode, 0);
+  } finally {
+    cleanup(projectName);
+  }
+})
+
+test('install - allows execution of allowed yarn classic dependencies', async (t: any) => {
```

Comment:
Much more explanation is gonna be necessary for this one. A big block comment with links preferably. I know what it's referring to but I'm still having trouble following the test steps :sweat_smile:

Reply:
Added a pointer to fixture README which contains explanation and a few links - does that suffice? If not, would you be able to suggest missing parts?